cli/run_test.go

```diff
@@ -0,0 +1,178 @@
+		}
+		assert.Equal(t, "morty_smith", result, fmt.Sprintf("Expected 'printenv | morty_smith' in the output, got 'printenv | %s'", result))
+	})
+
+	t.Run("Run with env variable in command line has priority over the one in the env file", func(t *testing.T) {
+		t.Parallel()
+
+		envFile, err := createEnvFile(t.TempDir(), []byte("CHARACTER=rick_sanchez"))
+		defer os.Remove(envFile)
+		if err != nil {
+			t.Fatalf("Failed to write env file: %v", err)
+		}
+
+		// Set environment variables
+		envVars := []string{"CHARACTER=morty_smith"}
+		buff, msg, err := execRunCommand(&envVars, "run", "--env-file", envFile, "--", "printenv", "CHARACTER")
+		if err != nil {
+			t.Fatalf(msg, err)
+		}
+
+		result, found := findEnvVariableInLogs(&buff)
+		if !found {
+			t.Fatalf("Expected printenv in the output: %s", buff.String())
+		}
+		assert.Equal(t, "morty_smith", result, fmt.Sprintf("Expected 'printenv | morty_smith' in the output, got 'printenv | %s'", result))
+	})
+
+	t.Run("Run with env variable in local env vars has priority over the one in the env file", func(t *testing.T) {
+		// Do not use t.Parallel() here as we are manipulating the environment!
+
+		// Set environment variables
+		os.Setenv("CHARACTER", "morty_smith")
+		defer os.Unsetenv("CHARACTER")
+
+		envFile, err := createEnvFile(t.TempDir(), []byte("CHARACTER=rick_sanchez"))
+		defer os.Remove(envFile)
+		if err != nil {
+			t.Fatalf("Failed to write env file: %v", err)
+		}
+
+		buff, msg, err := execRunCommand(&[]string{}, "run", "--env-file", envFile, "--", "printenv", "CHARACTER")
+		if err != nil {
+			t.Fatalf(msg, err)
+		}
+
+		result, found := findEnvVariableInLogs(&buff)
+		if !found {
+			t.Fatalf("Expected printenv in the output: %s\n\n", buff.String())
+		}
+		assert.Equal(t, "morty_smith", result, fmt.Sprintf("Expected 'printenv | morty_smith' in the output, got 'printenv | %s'", result))
+	})
+}
+
+func execRunCommand(envVars *[]string, arg ...string) (bytes.Buffer, string, error) {
+	// Create a context with a timeout to ensure the process doesn't run indefinitely
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// add the api key to the arguments so the command can run without `dispatch login` being run first
+	arg = append(arg[:1], append([]string{"--api-key", "00000000"}, arg[1:]...)...)
+
+	// Set up the command
+	cmd := exec.CommandContext(ctx, dispatchBinary, arg...)
+
+	if len(*envVars) != 0 {
+		// Set environment variables
+		cmd.Env = append(os.Environ(), *envVars...)
+	}
+
+	// Capture the standard error
+	var errBuf bytes.Buffer
+	cmd.Stderr = &errBuf
+
+	// Start the command
+	if err := cmd.Start(); err != nil {
+		return errBuf, "Failed to start command: &v", err
+	}
+
+	// Wait for the command to finish or for the context to timeout
+	if err := cmd.Wait(); err != nil {
+		// Check if the error is due to context timeout (command running too long)
+		if ctx.Err() == context.DeadlineExceeded {
```

Comment:
The context error will always carry information about this being a timeout, could we simplify the code by removing this special case?

Reply:
It will do `exit 1` on `printenv ENV_VARIABLE`, so I've added this check to be sure that this command exited _properly_ and we can read `stderr` now to make sure `ENV_VARIABLE` has correct value (when `err` is `nil`).

Reply:
🤔 I'm not sure I understand the condition in which we would get a timeout, but if you know it's important here it's fine (maybe worth a comment to record that context).